### PR TITLE
Add contributing page describing the OMERO.server Java components lifecycle

### PR DESCRIPTION
**Pull request by sbesson**
This PR aims to backport most of the deployment infrastructure introduced for OMERO.server 5.6.5 and cover the lifecycle of the omero-* components using the Gradle build system.

Includes a short description of the standard layout as well as the release process using GitHub actions for SNAPSHOT/release deployment

Do we want to add something about the Javadoc ?

### CODE REVIEW

**Review by joshmoore — APPROVED** [review]
https://ome-contributing--7.org.readthedocs.build/en/7/omero-gradle-development.html?highlight=gradle seems fine. I was slightly confused by the mismatch between the page header and the file name but I can certainly live with it.

**Comment by sbesson** [comment]
You are right, there was a mismatch as the name of the page reflected the build system while the heading reflected the components. [3e7d6f9](https://github.com/ome/ome-contributing/pull/7/commits/3e7d6f9f7f53d85f43ccc27ba4ebe5b5ee70cd08) proposes to unify to the latter.

**Comment by jburel** [comment]
Note that the process also applies to https://github.com/ome/omero-gradle-plugins

**Comment by sbesson** [comment]
> Note that the process also applies to https://github.com/ome/omero-gradle-plugins

So would you be in favor or going in the opposite direction of [3e7d6f9](https://github.com/ome/ome-contributing/pull/7/commits/3e7d6f9f7f53d85f43ccc27ba4ebe5b5ee70cd08) and aim for  this page to describe the lifecycle of all OME Gradle components?

**Comment by jburel** [comment]
> > Note that the process also applies to https://github.com/ome/omero-gradle-plugins
> 
> So would you be in favor or going in the opposite direction of [3e7d6f9](https://github.com/ome/ome-contributing/pull/7/commits/3e7d6f9f7f53d85f43ccc27ba4ebe5b5ee70cd08) and aim for this page to describe the lifecycle of all OME Gradle components?

yes since it is not limited to the server components

**Comment by sbesson** [comment]
See https://ome-contributing--7.org.readthedocs.build/en/7/gradle-development.html for the new version of the contributing page

File: docs/gradle-development.rst [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+Release process
+---------------
+
+To  make a new component release, after merging all contributions, on the `master`
+branch, the version defined in :file:`build.gradle` must be moved out of SNAPSHOT
+and set to the target version. Release notes described the major changes should also
+be added to :file:`CHANGELOG.md`.
+
+After committing the changes, a PGP-signed tag must be created for the released version
```
Comment (jburel) [inline docs/gradle-development.rst:77]: This needs to be unified with https://github.com/ome/ome-contributing/pull/8/. One indicates ``GPG`` one ``PGP`` cc @will-moore

Comment (sbesson) [reply docs/gradle-development.rst:77]: The usage of `PGP` is historical

```
(base) sbesson@Sebastiens-MacBook-Pro ome-contributing % git grep PGP
docs/ansible-development.rst:A PGP-signed tag of form `x.y.z` should be created for the released version
docs/cpp-development.rst:- have a valid PGP key for signing the tags
docs/cpp-development.rst:A PGP-signed tag should be created for the released version e.g.
docs/gradle-development.rst:After committing the changes, a PGP-signed tag must be created for the released version
docs/java-development.rst:- have a valid PGP key for signing the tags and the JARs
docs/java-development.rst:Additionally, a PGP-signed tag should be created for the released version e.g.
docs/tools-maintenance.rst:- have a valid PGP key for signing the tags
docs/using-git.rst:If you have a PGP key for signing commits and tags, you may want to add it as
docs/using-git.rst:    git config --global user.signingkey YOUR_PGP_KEY_ID
```

and effectively inherited from as far as the OME Trac wiki days. 

No objection to switching to GPG everywhere as this is effectively the technology we are using these days for commit/artifact signatures.

**Review by jburel — APPROVED** [review]
``GPG`` update can be done in a follow-up PR.

**Pull request merged**